File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # Boyer-Moore Voting Algorithm
        candidate = None
        count = 0 
        for num in nums :
            if count == 0 :
                candidate = num
                count = 1
            elif candidate == num :
                count += 1 
            else :
                count -= 1 
        if nums.count(candidate) > len(nums) // 2 :
            return candidate
        return -1

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        Integer candidate = null ; 
        int count = 0 ; 
        for (int num : nums)
        {
            if (count == 0)
            {
                candidate = num ;
                count = 1 ;  
            }
            else if (candidate == num) 
            {
                count++ ; 
            }
            else {
                count-- ; 
            }
        }

        return candidate ; 
    }
}